app/models/user.rb

```diff
@@ -143,6 +149,36 @@ def web_push_subscription(session)
     session.web_push_subscription.nil? ? nil : session.web_push_subscription.as_payload
   end
 
+  def approved?
+    approved_at.present?
+  end
+
+  def active_for_authentication?
+    super && (approved? || !Setting.need_approval)
+  end
+
+  def inactive_message
+    if !approved? && Setting.need_approval
+      :not_approved
+    else
+      super
+    end
+  end
+
+  def request_approval
+    if Setting.need_approval
+      User.admins.each do |admin|
+        UserMailer.new_user_waiting_for_approval(admin, self, 'hogehoge').deliver_later
+      end
+    else
+      approve
+    end
+  end
```

Comment:
we would generally put code like this into a RequestApprovalService and the RegistrationsController instead of using a callback.